### PR TITLE
Update AML Deployment

### PR DESCRIPTION
**Pull request by mrwyattii**
Recent feature additions like load balanced model replicas were not properly tested on the AML deployments. Updating the AML deployments and making minor improvements/fixes to work with multiple model replicas.

Co-Author: @DannyBruno
Co-Author: @srsaggam

### CODE REVIEW

File: mii/aml_related/templates.py [hunk 1]
```diff
@@ -164,9 +164,9 @@
 
 RUN /opt/miniconda/envs/amlenv/bin/pip install -r "$BUILD_DIR/requirements.txt" --extra-index-url https://download.pytorch.org/whl/cu113 && \
     /opt/miniconda/envs/amlenv/bin/pip install azureml-inference-server-http && \
-    /opt/miniconda/envs/amlenv/bin/pip install git+https://github.com/microsoft/deepspeed.git && \
-    /opt/miniconda/envs/amlenv/bin/pip install git+https://github.com/microsoft/DeepSpeed-MII.git && \
-    /opt/miniconda/envs/amlenv/bin/pip install transformers==4.21.2
+    /opt/miniconda/envs/amlenv/bin/pip install git+https://github.com/microsoft/DeepSpeed.git@mrwyattii/fix-multi-node-checks && \
```
Comment (mrwyattii) [inline mii/aml_related/templates.py:167]: Make default branch

File: mii/aml_related/templates.py [hunk 2]
```diff
@@ -164,9 +164,9 @@
...
+    /opt/miniconda/envs/amlenv/bin/pip install git+https://github.com/microsoft/DeepSpeed-MII.git@mrwyattii/update-AML-deployment && \
```
Comment (mrwyattii) [inline mii/aml_related/templates.py:168]: Make default branch

File: mii/deployment.py [hunk 3]
```diff
@@ -182,13 +196,14 @@ def _allocate_processes(hostfile_path, tensor_parallel, num_replicas):
     allocated_num = 0
     for host, slots in resource_pool.items():
         available_on_host = slots
-        while available_on_host >= tensor_parallel:
+        while True:  #available_on_host >= tensor_parallel:
```
Comment (mrwyattii) [inline mii/deployment.py:199]: Move to when we standing up the server (i.e., called from score.py)

File: mii/models/providers/huggingface.py [hunk 4]
```diff
@@ -187,9 +188,12 @@ def hf_provider(model_path, model_name, task_name, mii_config):
         return load_with_meta_tensor(model_path, model_name, task_name, mii_config)
     else:
         device = get_device(load_with_sys_mem=mii_config.load_with_sys_mem)
+        import os
```
Comment (mrwyattii) [inline mii/models/providers/huggingface.py:191]: remove debug prints

File: mii/models/score/score_template.py [hunk 5]
```diff
@@ -24,24 +24,32 @@ def init():
...
-                  task_name,
-                  model_name,
-                  model_path,
-                  ds_optimize=configs[mii.constants.ENABLE_DEEPSPEED_KEY],
-                  ds_zero=configs[mii.constants.ENABLE_DEEPSPEED_ZERO_KEY],
-                  ds_config=configs[mii.constants.DEEPSPEED_CONFIG_KEY],
-                  mii_configs=configs[mii.constants.MII_CONFIGS_KEY],
-                  lb_config=configs.get(mii.constants.LOAD_BALANCER_CONFIG_KEY,
-                                        None))
+    start_server = True
```
Comment (mrwyattii) [inline mii/models/score/score_template.py:27]: Add comments explaining the reasoning here

File: mii/server.py [hunk 6]
```diff
@@ -38,6 +40,8 @@ def __init__(self,
                  ds_config=None,
                  mii_configs={},
                  lb_config=None):
+        import os
+        print("SERVER PID", os.getpid())
```
Comment (mrwyattii) [inline mii/server.py:44]: remove debug statements

**Review by jeffra — APPROVED** [review]
🎉

**Review by srsaggam — APPROVED** [review]

**Pull request merged**